frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx

```diff
@@ -185,6 +186,29 @@ const ManageKServeModal: React.FC<ManageKServeModalProps> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [createDataInferenceService.name, setKserveNameDesc]);
 
+  React.useEffect(() => {
+    if (modelDeployPrefillInfo) {
+      const isDataReady =
+        !!kServeNameDesc.name &&
+        Array.isArray(servingRuntimeTemplates) &&
+        servingRuntimeTemplates.length > 0 &&
+        (!projectContext || connectionsLoaded) &&
+        (podSpecOptionsState.acceleratorProfile.loaded ||
+          podSpecOptionsState.hardwareProfile.profilesLoaded);
+
+      setIsLoading(!isDataReady);
+    }
+  }, [
+    kServeNameDesc.name,
+    servingRuntimeTemplates,
+    projectTemplates,
+    projectContext,
+    connectionsLoaded,
+    modelDeployPrefillInfo,
+    podSpecOptionsState.acceleratorProfile.loaded,
+    podSpecOptionsState.hardwareProfile.profilesLoaded,
+  ]);
+
```

Comment:
Same here, we can avoid using state for `isLoading` if its value is just derived from other things we know at render time. It can just be a regular variable with no useEffect.

```ts
const isDataReady = !modelDeployPrefillInfo ||
  !!kServeNameDesc.name &&
  Array.isArray(servingRuntimeTemplates) &&
  servingRuntimeTemplates.length > 0 &&
  (!projectContext || connectionsLoaded) &&
  (podSpecOptionsState.acceleratorProfile.loaded ||
    podSpecOptionsState.hardwareProfile.profilesLoaded)
```

Reply:
Ah - this is similar to what @manaswinidas was saying with the useMemo - we can useMemo this as well if we want, I think it's most important to avoid introducing useState/useEffect here if we can avoid them. But I don't think useMemo gains us much since React must compare the values of all its dependencies every render anyway and the logic inside is about the same amount of boolean comparison, so we may as well just not memoize it IMO. I don't feel strongly about that part though.